### DISCUSSION TITLE
Skeleton Next - Update 2 - Themes & Plugin

### DISCUSSION THREAD

**Discussion by endigo9740** [skeletonlabs/skeleton#2452] [Announcements]
Hey everyone, Chris here from the Skeleton crew! As part of our [proposal and RFC](https://github.com/skeletonlabs/skeleton/discussions/2375) and [ongoing series of progress updates](https://github.com/skeletonlabs/skeleton/discussions/2424) regarding Skeleton v3 (aka "Next"). We're excited to share some exciting new information today, including updates to the new documentation website, as well as talking through major changes coming to both Skeleton Themes and the Tailwind plugin.

As with the previous updates, we've provide a quick summary post below, as well as videos providing a more in-depth walkthrough on YouTube.

## YouTube Walkthroughs

- [Documentation Walkthrough](https://youtu.be/l-EIqrLkylA) (~20 mins)
- [Themes & Tailwind Plugin Walkthrough](https://youtu.be/2ap1wC_Jtto) (~40 mins)

## Useful Links

- [Preview the new documentation website](https://next.skeleton.dev/docs/colors)
- [Website source code on GitHub](https://github.com/skeletonlabs/skeleton/tree/next/sites/next.skeleton.dev)
- [Tailwind plugin source code on GitHub](https://github.com/skeletonlabs/skeleton/tree/next/packages/skeleton/src/plugin)

## Documentation Updates

![Screenshot 2024-01-29 at 12 00 21 PM](https://github.com/skeletonlabs/skeleton/assets/1509726/1911c2a2-0f79-490b-83bd-a3b12c180a34)

As mentioned in our proposal, the new documentation website utilizes [Astro](https://astro.build/). A meta-framework with a focus on server-side generation. This [provides a number of awesome features](https://docs.astro.build/en/concepts/why-astro/#features) out of the box.

While the new iteration of the Skeleton documentation website remains fairly barebones (ha, get it!), we have began the process of implementing several key improvements. This includes:

- A much flatter navigation
- Clearer separation between topics and subject matter, such as onboarding, design systems, etc.
- Early hints of the new design aesthetic are present as we begin utilizing Skeleton buttons, cards, etc
- Improved visual aids for explaining core concepts, such as color pairings
- As well as an initial draft of the "Cookbook" section, providing simple recipes for using Skeleton features

That aside, things are still a work in progress:

- We're using an early-stage test theme, so this does not represent the final design
- We do not yet have support for proper code blocks or syntax highlighting for code snippets
- Most usage information has not yet been written or included yet
- Many key sections are missing, such as: installation, themes, and component pages
- And many other omissions, such as poor mobile responsive support

These items will be the focus of the [next major wave of updates](https://github.com/skeletonlabs/skeleton/issues/2353), as we prepare the docs to support the new Svelte and React components. Additionally, progress will soon begin on the new [standalone theme generator](https://github.com/skeletonlabs/skeleton/issues/1142), which will enable us to preview several production-grade themes.

All this aside, we're pleased with the direction the documentation is heading, as we very much looking forward to having more to share with you in the coming weeks.

## Theme Updates

![Screenshot 2024-01-29 at 1 01 38 PM](https://github.com/skeletonlabs/skeleton/assets/1509726/1d7b2d1c-2d3f-4d89-a62d-0860153f3d0a)

You can compare the changes to the theme format here:

- [v2 Theme Format](https://github.com/skeletonlabs/skeleton/blob/dev/packages/plugin/src/tailwind/themes/skeleton.ts)
- [v3 Theme Format](https://github.com/skeletonlabs/skeleton/blob/next/packages/skeleton/src/plugin/themes/next.ts)

Themes represent the absolute foundation for Skeleton. The settings they house affect nearly every portion of the visual aesthetic and design systems within Skeleton, and handle configuration for a number of critical features.

As part of Next, we've overhauled themes to provide a number of new settings for the following features:

- Colors have been greatly expanded, including the new 950 shade, plus full support for contrasting tones per _every_ shade.
- Typography has been completely reimagined to support a full typography scale, as well as greater range of customization for base text styles, heading styles, and more.
- We've included a new Spacing system that affects padding, margin, width/height, and more. This now includes a simple yet powerful scaling factor to easily provide a base scale per theme.
- New CSS variables have been added to define background colors and various edge/border styles, with more planned for the future - including support for body background images.

Similar to the documentation above, these updates and changes are not yet complete. However, even in these early stages you can already start to imagine the depth of customization. While the number of changes can initially be a bit overwhelming, please be aware that we will be providing a very simple interface for creating themes using the new theme generator. This will include simple color pickers, range sliders, presets, and more. Making it extremely easy to curate your custom theme. Likewise, expect more updates on Themes in the future. Especially once the new generator is available to preview.

## Tailwind Plugin Updates

![Screenshot 2024-01-29 at 1 03 20 PM](https://github.com/skeletonlabs/skeleton/assets/1509726/d08a0c74-5522-4b51-9b42-331b8ad9eeba)

We've not just updated the [Skeleton Tailwind plugin](https://github.com/skeletonlabs/skeleton/tree/next/packages/skeleton/src/plugin), but in many cases have done a complete rewrite of the scripts and styles for many of the core features of Skeleton:

- The plugin's structure has been completely overhauled to be flatter and easier to navigate and expand over time
- We've leaned into supporting more of the features provide by Tailwind and it's extensive plugin system
- In many cases we're now properly extending Tailwind, rather than implement unique new classes on top of it
- [v2's variant styles](https://www.skeleton.dev/docs/variants) have now been replaced with [v3's preset system](https://next.skeleton.dev/docs/presets). Providing a robust set of utility classes that can be mixed and matched for powerful customization
- Nearly every Tailwind Component (aka "Elements") has been rewritten to support the new theme system, provide more features, while retaining the simple and familiar class names and syntax.
- [Table](https://next.skeleton.dev/tailwind/tables) and [Form](https://next.skeleton.dev/tailwind/forms) components specifically have received a number of quality of life improvements.

Like the other updates today, the Tailwind plugin is not yet finished. However, the foundation is now in place to allow us to move forward with other critical portions of the v3 update. This includes further improvements to the doc website, the first draft of the new theme generator, and of course starting to port the new Svelte 5 and React components. You can expect more updates on these features in the coming weeks.

In the meantime, if you have any questions or comments, please reach out in the comments below. We love hearing from you, our wonderful community!

**Comment by adminy** [comment]
What happened to the Table, Form & Preview links? Actually pretty much all Links.

**Reply by Sarenor** [reply]
Currently the preview site is undergoing a technical migration to a different hosting provider. Give it another day, maybe 2

**Reply by endigo9740** [reply]
Sorry about that @adminy I shared on Discord but didn't have a great way to communicate it here. The TL;DR is we're migrating hosting back to Vercel (where we typically host projects), but we've identified an error between Astro's linter and Svelte 5. Svelte 5 support is considered experimental (for good reason), so this is not unexpected. But it is hampering our ability to resolve the issue.

We're communicating with the Astro team and hope to have a fix asap. For now, you are more than welcome to clone down the project and run things locally.

- Switch to the `next` branch
- Navigate to the doc site in: `/sites/next.skeleton.dev`
- Run `pnpm i`, then `pnpm run dev` to start a local server

Otherwise we'll have this back up and running asap!

**Reply by endigo9740** [reply]
@adminy we're still working through a domain issue - but you can at least preview the latest changes here:

*EDIT*

The site is live again!
https://next.skeleton.dev/
